profiles/rawls-service-perimeters/terraform/rawls-service-perimeters/perimeters.tf

```diff
@@ -45,4 +45,7 @@ resource "google_access_context_manager_service_perimeter" "service-perimeter" {
       status[0].resources
     ]
   }
+  provisioner "local-exec" {
+    command = "./create-sam-perimeter-resource.sh ${var.terra_environment} ${each.key} ${var.access_policy_name} ${each.value.sam_resource_owner_email}"
```

Comment:
> By default, provisioners that fail will also cause the Terraform apply itself to fail. 
We updated the script so that it will fail unless the curl command to create the resource returns a 201, 204 or 409. These are the expected status codes, and any other codes would indicate that something went wrong when creating the Sam resource. In this case, the Terraform apply will also fail and taint the resource which would reflect the state of the Sam resource as well. Fixing both the tainted Terraform resource and the Sam service-perimeter resource would require manual effort, which seems okay to me but I welcome input on that.

Reply:
I know there's an `untaint` command, although I don't know how we would go about running it. We might need some sort of DevOps assistance. What I'm envisioning for a failed run is basically this:
1. Terraform apply fails because the script fails and the Terraform resource is tainted
2. Investigate why the script failed (likely some unexpected status code from Sam)
3. Fix the service-perimeter resource in Sam
4. Once the reason the script failed is resolved, untaint the Terraform resource

If the entire Terraform apply didn't fail when the script failed, I'd be worried about silent failures from the script that leave the service perimeter in bad shape without us knowing that something went wrong when we were creating it. One caveat to that would be if the script fails when _updating_ the perimeter and taints the entire Terraform resource because of a transient error that has no effect on the functionality of the perimeter.